iroh-net/src/endpoint.rs

```diff
@@ -0,0 +1,542 @@
+use std::{
```

Comment:
Can we call this `magic_endpoint.rs` please? I am always confused if the structs don't match the filename 😅